app/models/concerns/sequenced.rb

```diff
@@ -32,20 +40,50 @@ def generate_sequential_id
         end
       end
 
-      # NOTE: If the application was unable to aquire the lock, the block returns false
-      raise(SequenceError, 'Unable to aquire lock on the database') unless result
+      # NOTE: If the application was unable to acquire the lock, the block returns false
+      raise(SequenceError, 'Unable to acquire lock on the database') unless result
+
+      result
+    end
+
+    def generate_organization_sequential_id
+      result = Invoice.with_advisory_lock(
```

Comment:
This logic seems specific to the Invoice model, and not related to the exciting sequence logic. What do you think of adding it directly in the model? This module is shared between multiple models and is generic, I think it should not contain specific model logic

Reply:
make sense 👍